bool need_initial_estimate
geometry_msgs/Pose current_pose
---
geometry_msgs/Pose optimized_pose